larpandoracontent/LArControlFlow/MasterAlgorithm.cc

```diff
@@ -418,6 +418,29 @@ StatusCode MasterAlgorithm::RunCosmicRayHitRemoval(const PfoList &ambiguousPfos)
     {
         const ClusterList clusterList(pPfoToDelete->GetClusterList());
         const VertexList vertexList(pPfoToDelete->GetVertexList());
+
+        // ATTN: If an ambiguous pfo has been stitched, reset the calo hit positions in preparation for subsequent algorithm chains
+        if (pPfoToDelete->GetPropertiesMap().count("X0"))
```

Comment:
Using the presence of a string in the map means this algorithm relies on the implementation in other parts of the code. I wonder whether we could instead go for a single definition in e.g. `LArPfoHelper::HasPfoBeenStitched(const Pfo *const);` and use this in *all* cases where we want to know this.

One way to avoid this would be to just look at the hits, but this means looping over all the hits, all the time. Alternatively, could push the actual string into xml at the point of adding the property and when checking it.

I think I'm in favour of consolidating this functionality into one place for reduced maintenance. What do you think?

Reply:
Hi John, 

This is a good point, I like the idea of having the additional LArPfoHelper function as it will reduce complications in the future.  I'll start work on adding that now.  

I'm not sure I understand what you mean by pushing the string into xml at the point of adding the property and when checking it though.  Could you clarify as I'm probably just misunderstanding something obvious!

Reply:
Hi Steve,

Sorry this wasn't clear (unless you're a mind reader!). Basically I'm rarely comfortable when we have literals, especially when we're relying on filling the map via a string literal key in one algorithm and then accessing it, via the same literal, in another. It can be difficult to maintain when changes are made in future.

One idea is to move the literal into just one place (or maybe one place for setting, one place for accessing) e.g. in a helper, to ease maintenance. Another idea would be to set and read the map using a string that is read in via an XML config file, to provide the key. This still means setting and accessing can get out of sync, but it can be recovered via config, rather than being irrevocably compiled in. I hope this clarifies things!

It looks like LArStitchingHelper would actually be the best place to consolidate this. If every use/check goes through this helper, then it's much easier to maintain.

What do you think?

Reply:
Hi John,
Thanks for the feedback, sorry for the slow response.  Ah yes, now I understand about defining the string to be set via xml.  I'll add these additional functions to the Stitching helper for now to keep things under control as you suggest.  
Cheers